diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt

```diff
@@ -62,13 +68,16 @@ class TrailingCommaRule(private val configRules: List<RulesConfig>) : Rule("trai
         emitWarn = emit
         isFixMode = autoCorrect
 
-        val commonConfig by configRules.getCommonConfiguration()
         if (commonConfig.kotlinVersion >= ktVersion) {
             val (type, config) = when (node.elementType) {
                 VALUE_ARGUMENT_LIST -> Pair(VALUE_ARGUMENT, configuration.getParam("valueArgument"))
                 VALUE_PARAMETER_LIST -> Pair(VALUE_PARAMETER, configuration.getParam("valueParameter"))
                 INDICES -> Pair(REFERENCE_EXPRESSION, configuration.getParam("referenceExpression"))
-                WHEN_ENTRY -> Pair(WHEN_CONDITION_WITH_EXPRESSION, configuration.getParam("whenConditions"))
+                WHEN_ENTRY -> Pair(when {
+                    node.hasChildOfType(WHEN_CONDITION_WITH_EXPRESSION) -> WHEN_CONDITION_WITH_EXPRESSION
```

Comment:
I think it's better to take the last child of one of these types. They can be mixed in one when entry, I think.

Reply:
Fixed